### PR TITLE
Move requirements specification to PEP 0621

### PR DESCRIPTION
**Pull request by FranzForstmayr**
I tried to clean up the repository by using `pyproject.toml` as dependency specification. Let's see if pipeline runs without errors.

### CODE REVIEW

**Comment by FranzForstmayr** [comment]
@jhillairet 
Do you get more output from readthedocs? 
I don't know why the build status shows python3.7 when I select python3.8.

**Comment by FranzForstmayr** [comment]
Maybe we should change to the official PEP 621 file format in the future:
https://peps.python.org/pep-0621

I gave it a first try:
```toml
[project]
name = "scikit-rf"
version = "0.23.1"
description = "Object Oriented Microwave Engineering"
requires-python = ">=3.7"
authors = [
    {name="Alex Arsenovic", email="alexanderarsenovic@gmail.com"}
]
license = {text="LICENSE.txt"}
readme = "README.md"

keywords = [
    "engineering", 
    "electronics", 
    "microwave-engineering", 
    "radio-frequency", 
    "touchstone"]

classifiers = [
    "Development Status :: 5 - Production/Stable",
    "Intended Audience :: Education",
    "Intended Audience :: Science/Research",
    "License :: OSI Approved :: BSD License",
    "Operating System :: OS Independent",
    "Programming Language :: Python",
    "Topic :: Communications :: Ham Radio",
    "Topic :: Scientific/Engineering"]

dependencies = [
    "numpy ~=1.21",
    "scipy ~=1.7",
    "pandas ~=1.1",
]

[project.optional-dependencies]
test = [
    "pytest >=7",
    "coverage >=6",
    "flake8 >=5",
    "pytest-cov >=3",
    "nbval >=0.9"
]

plot = [
    "matplotlib ~=3.5"
]

xlsx = [
    "openpysl ~=3.0"
]

netw = [
    "networkx ~= 2.0"
]

visa = [
    "PyVISA ~= 1.12"
]

docs = [
    "ipython >=7",
    "ipykernel >=6.15",
    "ipywidgets >=8.0.2",
    "python-ivi >=0.14.9",
    "jupyter-client >=7.3.5",
    "sphinx-rtd-theme >=1.0",
    "sphinx >=4",
    "nbsphinx >= 0.8.9"
]

[project.urls]
homepage = "https://scikit-rf.org/"
repository = "https://github.com/scikit-rf/scikit-rf"
documentation = "https://scikit-rf.readthedocs.io/en/latest/"

[build-system]
requires = [
  "setuptools >= 40.9.0",
  "wheel",
]
build-backend = "setuptools.build_meta"

[tool.setuptools]
packages = ["skrf"]
license-files = ["LICENSE.txt"]
```

But it's not really supported by setuptools yet.

**Comment by FranzForstmayr** [comment]
> @jhillairet Do you get more output from readthedocs? I don't know why the build status shows python3.7 when I select python3.8.

@jhillairet, gentle ping :)

**Comment by jhillairet** [comment]
> > @jhillairet Do you get more output from readthedocs? I don't know why the build status shows python3.7 when I select python3.8.
> 
> @jhillairet, gentle ping :)

oups sorry, I missed that. I'll have a look asap

File: readthedocs.yml [hunk 1]
```diff
@@ -1,16 +1,9 @@
-# .readthedocs.yml
-# Read the Docs configuration file
-version: 2
```
Comment (jhillairet) [inline readthedocs.yml:3]: is removing this line goes back to V1 ? (and is it a problem?)

File: requirements.txt [hunk 2]
```diff
@@ -1,8 +0,0 @@
-matplotlib
-scipy
-numpy>=1.19.5
-pandas
-nbsphinx
-networkx
-xlwt
```
Comment (jhillairet) [inline requirements.txt:7]: poetry newbie question : how does pip and users of pip will deal without this requirement file? (cf https://pip.pypa.io/en/stable/user_guide/#requirements-files)

File: setup.py [hunk 3]
```diff
@@ -1,58 +0,0 @@
-#!/usr/bin/env python
-
-from setuptools import setup, find_packages
-from distutils.core import Extension
-
```
Comment (jhillairet) [inline setup.py:5]: Again, newbie question. I understand that poetry uses pyproject.toml to replace setup.py, requirements.txt, etc. However how do I update the pip package once a new version is released? I use `twine upload dist/*` to do that currently.

**Comment by jhillairet** [comment]
> @jhillairet Do you get more output from readthedocs? I don't know why the build status shows python3.7 when I select python3.8.

A more recent PR (https://readthedocs.org/projects/scikit-rf/builds/18033083/) does not have this problem. 

So I understand that the problem with [the current PR](https://readthedocs.org/api/v2/build/18024126.txt) is the fact that RTD does not find the removed config files... It also complains about the missing of [.readthedocs.yaml](https://docs.readthedocs.io/page/config-file/v2.html) (which is still here... Is it related to the version 1 vs 2 of the yaml file?)

File: readthedocs.yml (hunk 1 above)
Comment (FranzForstmayr) [reply readthedocs.yml:3]: I'll have to try anyway, as this is not working obviously :D

File: requirements.txt (hunk 2 above)
Comment (FranzForstmayr) [reply requirements.txt:7]: pip understands pyproject.toml and will install everything as usual.

**Comment by jhillairet** [comment]
Hi @FranzForstmayr, is everything is working now? What was the reason?

**Comment by FranzForstmayr** [comment]
The RTD config needs to be named `.readthedocs.yaml` and not `readthedocs.yaml`... Took me quite a while to figure out.

However I don't use poetry anymore, but the new [PEP 0621](https://peps.python.org/pep-0621/) Metadata format.
The reason is, that poetry wants to select one version of a package for all configurations, but it's hard to find a `numpy` version which works for Python 3.7 to Python 3.11. That's why poetry used always a version supported by Python 3.7 and built `numpy` from source for newer versions.  For PEP 0621 the dependency resolving is done by pip, which chooses one version for the selected Python environment.

**Review by jhillairet — APPROVED** [review]

**Comment by jhillairet** [comment]
OK for merging then

**Pull request merged**